### ISSUE TITLE
Validation and Test data

### ISSUE THREAD

**Issue by kr-sundaram** [devJWSong/transformer-translator-pytorch#2]
Hey, I found that in this repo, you have considered all the data as training data and did not considered to split your full dataset into train, validate, and test data.

Do you have plan to incorporate the splitting of the full dataset into these 3 dataset i.e. train, validate and test?

**Comment by devjwsong** [comment]
Thank you for your pointing out. Yes, as you can see, I'm using train data only for now.
Splitting dataset is important for preventing overfitting and evaluating the model quantitatively as you already know.

Actually, I'm a little embarrassed to say that the only reason why I didn't split the data is that I didn't know how to evaluate the translation model when I first started the project.
Of course now I know the perplexity or BLEU score can be implemented as automatic metrics but back then I was just a novice.
The part about validation will require further investigation, but there is clearly a plan to split the data as you say to enhance overall performance.

Thank you again and I will update my repo as soon as possible.

**Comment by kr-sundaram** [comment]
Maybe you will get some insights from the below repo for incorporating valid and test dataset in your repository.

https://github.com/bentrevett/pytorch-seq2seq/blob/master/6%20-%20Attention%20is%20All%20You%20Need.ipynb

**Comment by devjwsong** [comment]
I checked the training part in the repo you provided and I see that simply validation loss is used to evaluate the model.
I tried it for validation when I first built this project but for unknown reasons, valid loss kept increasing from the beginning while training loss decreased.
That's why I quit this method.
Maybe I have to try it again and check since the codes have been changed quite a lot.

Thank you for your help.
